src/main/python/systemml/classloader.py

```diff
@@ -51,8 +51,12 @@ def createJavaObject(sc, obj_type):
     try:
         return _createJavaObject(sc, obj_type)
     except (py4j.protocol.Py4JError, TypeError):
-        import imp
-        jar_file_name = os.path.join(imp.find_module("systemml")[1], "systemml-java", 'SystemML.jar')
+        import imp, fnmatch
+        jar_file_name = '_ignore.jar'
+        java_dir = os.path.join(imp.find_module("systemml")[1], "systemml-java")
+        for file in os.listdir(java_dir):
+            if fnmatch.fnmatch(file, 'systemml-*-incubating-SNAPSHOT.jar'):
+                jar_file_name = os.path.join(java_dir, file)
```

Comment:
Have we tested if this JAR will be distributed to the worker nodes on a cluster as well?